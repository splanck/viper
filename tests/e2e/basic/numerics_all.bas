10 PRINT "INT_POS="; INT(2.5#)
20 PRINT "INT_NEG="; INT(-2.5#)
30 PRINT "FIX_POS="; FIX(2.5#)
40 PRINT "FIX_NEG="; FIX(-2.5#)
50 PRINT "ROUND_TIE_LOW="; ROUND(2.5#)
60 PRINT "ROUND_TIE_HIGH="; ROUND(3.5#)
70 PRINT "CINT_POS="; CINT(2.5#)
80 PRINT "CLNG_NEG="; CLNG(-2.5#)
90 PRINT "CSNG="; CSNG(1.5#)
100 PRINT "CDBL="; CDBL(1.5#)
110 PRINT "VAL_FLOAT="; VAL("  -1.5E1X")
120 PRINT "VAL_NONNUM="; VAL("abc")
130 PRINT "STR_POS="; STR$(123)
140 PRINT "STR_NEG="; STR$(-42)
150 PRINT "DIV="; 3 / 2
160 PRINT "IDIV="; -7 \ 3
170 PRINT "MOD_NEG="; -3 MOD 2
180 PRINT "POW="; 2 ^ -3
