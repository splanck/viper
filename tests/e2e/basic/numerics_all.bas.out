INT_POS=2
INT_NEG=-3
FIX_POS=2
FIX_NEG=-2
ROUND_TIE_LOW=2
ROUND_TIE_HIGH=4
CINT_POS=2
CLNG_NEG=-2
CSNG=1.5
CDBL=1.5
VAL_FLOAT=-15
VAL_NONNUM=0
STR_POS=123
STR_NEG=-42
DIV=1.5
IDIV=-2
MOD_NEG=-1
POW=-2
